lib/ruby_lsp/listener.rb

```diff
@@ -42,7 +49,10 @@ def add_listener(listener)
       def listener_events(&block)
         current_methods = instance_methods
         block.call
-        @events = T.let(instance_methods - current_methods, T.nilable(T::Array[Symbol]))
+
+        (instance_methods - current_methods).each do |event|
+          T.must(Listener.event_to_listener_map[event]) << self
```

Comment:
```suggestion
          Listener.event_to_listener_map.fetch(event) << self
```

Reply:
I'm not really a fan of this `fetch` pattern to be honest. Unless there's consensus from the team that we should be using it, I prefer hash access.